Svc/FileDispatcher/FileDispatcher.cpp

```diff
@@ -0,0 +1,84 @@
+// ======================================================================
+// \title  FileDispatcher.cpp
+// \author tcanham
+// \brief  cpp file for FileDispatcher component implementation class
+// ======================================================================
+
+#include "Svc/FileDispatcher/FileDispatcher.hpp"
+#include "Fw/Types/StringUtils.hpp"
+
+namespace Svc {
+
+// ----------------------------------------------------------------------
+// Component construction and destruction
+// ----------------------------------------------------------------------
+
+FileDispatcher ::FileDispatcher(const char* const compName) : FileDispatcherComponentBase(compName) {
+    // disable entries
+    for (FwSizeType i = 0; i < Svc::FileDispatcherCfg::FileDispatchPort::MAX_FILE_DISPATCH_PORTS; i++) {
+        this->m_dispatchTable[i].enabled = false;
+    }
+}
+
+FileDispatcher ::~FileDispatcher() {
+    this->deinit();
+}
+
+void FileDispatcher ::configure(FileDispatcherEntry* entries, FwSizeType numEntries) {
+    FW_ASSERT(entries != nullptr);
+    FW_ASSERT(numEntries > 0);
+    FW_ASSERT(numEntries <= Svc::FileDispatcherCfg::FileDispatchPort::MAX_FILE_DISPATCH_PORTS);
+
+    // copy entries into table
+    for (FwSizeType i = 0; i < numEntries && i < Svc::FileDispatcherCfg::FileDispatchPort::MAX_FILE_DISPATCH_PORTS;
+         i++) {
+        this->m_dispatchTable[i] = entries[i];
+    }
+    // disable remaining entries
+    for (FwSizeType i = numEntries; i < Svc::FileDispatcherCfg::FileDispatchPort::MAX_FILE_DISPATCH_PORTS; i++) {
+        this->m_dispatchTable[i].enabled = false;
+    }
+}
+
+// ----------------------------------------------------------------------
+// Handler implementations for typed input ports
+// ----------------------------------------------------------------------
+
+void FileDispatcher ::fileAnnounceRecv_handler(FwIndexType portNum, Fw::StringBase& file_name) {
+    // determine file extension and dispatch accordingly
+
+    // walk table to find match
+    for (FwIndexType i = 0; i < Svc::FileDispatcherCfg::FileDispatchPort::MAX_FILE_DISPATCH_PORTS; i++) {
+        if (this->m_dispatchTable[i].enabled) {
+            if (Fw::StringUtils::substring_find(file_name.toChar(), file_name.length(),
```

Comment:
This will search for the pattern "fileExt" anywhere in the filename.   This really should do a `substring_find_last` (https://github.com/nasa/fprime/blob/1a097ade00279e7dc885c19de9117f9fa31001e8/Fw/Types/StringUtils.cpp#L77) and confirm the index is `file.length()` - `ext.length()`.

Finally, this is suffix driven, not extension driven as it is not enforcing "."